apollo-router/src/services/supergraph_service.rs

```diff
@@ -153,7 +154,26 @@ where
                 QueryPlannerContent::Plan { query, plan } => {
                     let can_be_deferred = plan.root.contains_defer();
 
-                    if let Some(err) = query.validate_variables(body, &schema).err() {
+                    if can_be_deferred && !req.originating_request.headers().get_all(ACCEPT).iter().filter_map(|value| value.to_str().ok()).flat_map(|value| value.split(',')
```

Comment:
I think this is a fine starting point but I'd suggest at least an issue-and-a-comment noting that this isn't a full spec complaint accept parser. (most importantly it will get confused if you have any "parameters" in accept like `accept: multipart/mixed; foo=bar, application/json`; less importantly it can get even more confused if any parameters use quoted strings, like `accept: application/json; someParam="foo, multipart/mixed, bar"`.)

(and yes, the way that parameters bind to types in Accept header involves semicolons that bind tighter than commas, unlike in English/French/etc)

Reply:
right, this is something I planned to fix too. I'm a bit disappointed that the http crate does not provide a spec compliant iterator over values for well known headers